File: src/Python/Parcial2/ActividadP2A4/empleadoporhoras.py
from empleado import Empleado

class EmpleadoPorHoras(Empleado):
    def __init__(self, nombre, apellido, nss, sueldoPorHoras, horasTrabajadas):
        super().__init__(nombre, apellido, nss)
        self.establecerSueldo(sueldoPorHoras)
        self.establecerHoras(horasTrabajadas)

    def establecerSueldo(self, sueldoPorHoras):
        if sueldoPorHoras < 0.0:
            self.sueldo = 0.0
        else:
            self.sueldo = sueldoPorHoras

    def obtenerSueldo(self):
        return self.sueldo

    def establecerHoras(self, horasTrabajadas):
        if horasTrabajadas >= 0.0 and horasTrabajadas <= 168.0:
            self.horas = horasTrabajadas
        else:
            self.horas = 0.0

    def obtenerHoras(self):
        return self.horas

    def ingresos(self):
        if self.obtenerHoras() <= 40:
            return self.obtenerSueldo() * self.obtenerHoras()
        else:
            return 40 * self.obtenerSueldo() + (self.obtenerHoras() - 40) * self.obtenerSueldo() * 1.5

    def __str__(self):
        return f"empleado por horas: {super().__str__()}\nsueldo por hora: ${self.obtenerSueldo()}; horas trabajadas: {self.obtenerHoras()}"

File: src/C#/Parcial2/ActividadP2A4/EmpleadoPorHoras.cs
using System;

public class EmpleadoPorHoras : Empleado
{
    private double sueldo;
    private double horas;

    public EmpleadoPorHoras(string nombre, string apellido, string nss, double sueldoPorHoras, double horasTrabajadas)
        : base(nombre, apellido, nss)
    {
        EstablecerSueldo(sueldoPorHoras);
        EstablecerHoras(horasTrabajadas);
    }

    public void EstablecerSueldo(double sueldoPorHoras) { sueldo = (sueldoPorHoras < 0.0) ? 0.0 : sueldoPorHoras; }
    public double ObtenerSueldo() { return sueldo; }
    public void EstablecerHoras(double horasTrabajadas) { horas = ((horasTrabajadas >= 0.0) && (horasTrabajadas <= 168.0)) ? horasTrabajadas : 0.0; }
    public double ObtenerHoras() { return horas; }

    public override double Ingresos()
    {
        if (ObtenerHoras() <= 40)
            return ObtenerSueldo() * ObtenerHoras();
        else
            return 40 * ObtenerSueldo() + (ObtenerHoras() - 40) * ObtenerSueldo() * 1.5;
    }

    public override string ToString()
    {
        return "empleado por horas: " + base.ToString() + "\nsueldo por hora: $" + ObtenerSueldo() + "; horas trabajadas: " + ObtenerHoras();
    }
}
